joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/frontend/XTypeRecovery.scala

```diff
@@ -99,16 +99,20 @@ abstract class XTypeRecoveryPass[CompilationUnitType <: AstNode](
     import io.joern.x2cpg.passes.frontend.XTypeRecovery.AllNodeTypesFromNodeExt
 
     def getFieldBaseTypes(fieldAccess: FieldAccess): Iterator[TypeDecl] = {
-      fieldAccess.argument(1) match
-        case x: Call if x.name == Operators.fieldAccess =>
-          cpg.typeDecl.fullNameExact(FieldAccess(x).referencedMember.getKnownTypes.toSeq*)
-        case x: Call if !x.name.startsWith("<operator>") =>
-          if (!x.typeFullName.matches(XTypeRecovery.unknownTypePattern.pattern.pattern()))
-            cpg.typeDecl.fullNameExact(x.typeFullName)
-          else
-            Iterator.empty
-        case x: Expression =>
-          cpg.typeDecl.fullNameExact(x.getKnownTypes.toSeq*)
+      fieldAccess.argumentOption(1) match
```

Comment:
Could we perhaps make this a `.map` instead? i.e. 
```scala
fieldAccess.argumentOption(1).map {
case x: Call if x.name == Operators.fieldAccess => ....
case x: Call if !x.name.startsWith("<operator>") => ...
}.getOrElse(Iterator.empty)
```

Reply:
More readable to have this instead of multiple layers of `match` expressions

Reply:
done